ament_mypy/ament_mypy/main.py

```diff
@@ -0,0 +1,257 @@
+             'null device'
+    )
+
+    # not using a file handle directly
+    # in order to prevent leaving an empty file when something fails early
+    parser.add_argument(
+        '--xunit-file',
+        help='Generate a xunit compliant XML file'
+    )
+    args = parser.parse_args(argv)
+
+    if args.xunit_file:
+        start_time = time.time()
+
+    filenames = get_files(args.paths)
+    if args.excludes:
+        filenames = [f for f in filenames
+            if os.path.basename(f) not in args.excludes]
+    if not filenames:
+        print('No files found', file=sys.stderr)
+        return 1
+
+    if args.config_file and not os.path.exists(args.config_file):
+        print("Could not find config file '{}'".format(args.config_file),
+            file=sys.stderr
+        )
+        return 1
+
+    normal_report, error_messages, exit_code = generate_mypy_report(
+        args.config_file,
+        filenames,
+        args.cache_dir
+    )
+
+    if error_messages:
+        print('mypy error encountered', file=sys.stderr)
+        print(error_messages, file=sys.stderr)
+        return exit_code
+
+    errors_parsed = get_errors(normal_report)
+
+    print('\n{} files checked'.format(len(filenames)))
+    if not normal_report:
+        print('No errors found')
+    else:
+        print('{} errors'.format(len(errors_parsed)))
+
+    print(normal_report)
+
+    print('\nChecked files:')
+    print(''.join(['\n* {}'.format(f) for f in filenames]))
+
+    # generate xunit file
+    if args.xunit_file:
+        folder_name = os.path.basename(os.path.dirname(args.xunit_file))
+        file_name = os.path.basename(args.xunit_file)
+        suffix = '.xml'
+        if file_name.endswith(suffix):
+            file_name = file_name[:-len(suffix)]
+            suffix = '.xunit'
+            if file_name.endswith(suffix):
+                file_name = file_name[:-len(suffix)]
+        testname = '{}.{}'.format(folder_name, file_name)
+
+        xml = get_xunit_content(errors_parsed, testname, filenames, time.time() - start_time)
+        path = os.path.dirname(os.path.abspath(args.xunit_file))
+        if not os.path.exists(path):
+            os.makedirs(path)
+        with open(args.xunit_file, 'w') as f:
+            f.write(xml)
+
+    return exit_code
+
+
+def generate_mypy_report(config_file: str,
+                         paths: List[str],
+                         cache_dir: str) -> Tuple[str, str, int]:
+    mypy_argv = []
+    mypy_argv.append('--cache-dir')
+    mypy_argv.append(str(cache_dir))
+    if cache_dir == os.devnull:
+        mypy_argv.append('--no-incremental')
+    if config_file is not None:
+        mypy_argv.append('--config-file')
+        mypy_argv.append(str(config_file))
+    if not os.environ.get('MYPYPATH') and os.environ.get('PYTHONPATH') and os.environ.get('COLCON_PREFIX_PATH'):
+        # Filter PYTHONPATHs by colcon prefixes
+        python_paths = os.environ['PYTHONPATH'].split(':')
```

Comment:
This needs to use a platform specific separator.

Same below.

Reply:
fixed in 538014f